packages/react/src/lib/renderComponent.tsx

```diff
@@ -238,22 +322,12 @@ const renderComponent = <P extends {}>(
     displayName, // does not affect styles, only used by useEnhancedRenderer in docs
   }
 
-  const resolvedStyles: ComponentSlotStylesPrepared = {}
-  const resolvedStylesDebug: { [key: string]: { styles: Object }[] } = {}
-  const classes: ComponentSlotClasses = {}
-
-  Object.keys(mergedStyles).forEach(slotName => {
-    resolvedStyles[slotName] = callable(mergedStyles[slotName])(styleParam)
-
-    if (process.env.NODE_ENV !== 'production' && isDebugEnabled) {
-      resolvedStylesDebug[slotName] = resolvedStyles[slotName]['_debug']
-      delete resolvedStyles[slotName]['_debug']
-    }
-
-    if (renderer) {
-      classes[slotName] = renderer.renderRule(callable(resolvedStyles[slotName]), felaParam)
-    }
-  })
+  const { resolvedStyles, resolvedStylesDebug, classes } = resolveStylesAndClasses(
```

Comment:
Is it better to have this inline or in a separate function (with strange in/out API)?

Reply:
Yea, these args are odd and leaky:

- `renderer` is only used for `renderer.renderRule`.  We should just take in a single function here and pass `renderer.renderRule` in as something like `renderStyles`
- `felaParam` should probably be something like `renderStylesParam`

Ideally, this would be a function which you could use with any style function and/or renderer.  Crafting it in a way where it isn't leaking fela implementation details would be ideal.

Reply:
`renderer.renderRule` -> `renderStyles`
`felaParam` removed - it is `renderStyle`'s responsibility